facade/src/main/java/org/jboss/pnc/facade/providers/ProductMilestoneProviderImpl.java

```diff
@@ -275,6 +281,34 @@ public void cancelMilestoneCloseProcess(String id) throws RepositoryViolationExc
         return new Page<>(pageIndex, pageSize, milestoneIds.size(), milestones);
     }
 
+    @Override
+    public ValidationResponse validateVersion(String productVersionId, String version) {
+        boolean matches = Pattern.matches(Patterns.PRODUCT_MILESTONE_VERSION, version);
+        ValidationResponse.Builder builder = ValidationResponse.builder().isValid(matches);
+        if (!matches) {
+            return builder.errorType(FORMAT)
+                    .hints(
+                            Lists.newArrayList(
+                                    "Allowed format consists of 2 or 3 numeric components (separated by a dot) followed by a string "
+                                            + "qualifier starting with a character, eg. 3.0.0.GA, 1.0.11.CR2.ER1, 3.0.CR2"))
+                    .build();
+        }
+        org.jboss.pnc.model.ProductMilestone duplicate = null;
+        if (productVersionId != null) {
```

Comment:
if you use `@Valid` then you don't need this check, I think.

Reply:
I mean, I don't mind having null there. If it is null, then only REGEX check will be done.

Reply:
Then why did you added `@NotBlank` to it in the DTO?

Reply:
That's a valid point :D. I forgot that NotBlank doesn't allow nulls. I wanted it to cover a case "if there is a string don't let it be empty". I'll add the @Valid there.